webapp/src/components/BuyParcelPage/BuyParcelPage.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Link } from 'react-router-dom'
+import { Container, Header, Grid, Button, Message } from 'semantic-ui-react'
+import Navbar from 'components/Navbar'
+import ParcelName from 'components/ParcelName'
+import Parcel from 'components/Parcel'
+import { walletType } from 'components/types'
+import { locations } from 'locations'
+
+import './BuyParcelPage.css'
+
+export default class BuyParcelPage extends React.PureComponent {
+  static propTypes = {
+    wallet: walletType,
+    x: PropTypes.isRequired,
+    y: PropTypes.isRequired,
+    isDisabled: PropTypes.bool.isRequired,
+    onConfirm: PropTypes.func.isRequired,
+    onCancel: PropTypes.func.isRequired
+  }
+
+  handleConfirm = () => {
+    const { publication, onConfirm } = this.props
+    onConfirm(publication)
+  }
+
+  render() {
+    const { wallet, x, y, publication, isDisabled, onCancel } = this.props
+    const { approvedBalance } = wallet
+
+    const isNotEnough = publication && approvedBalance < +publication.price
+
+    return (
+      <div className="BuyParcelPage">
+        <Navbar />
+        <Parcel x={x} y={y}>
+          {parcel => (
+            <React.Fragment>
+              <Container text textAlign="center">
+                <Header as="h2" size="huge" className="title">
+                  Buy LAND
+                </Header>
+                <span className="subtitle">
+                  You are about to buy&nbsp;
+                  <ParcelName parcel={parcel} />{' '}
+                  {publication ? (
+                    <React.Fragment>
+                      for{' '}
+                      <strong className="price">
+                        {(+publication.price).toLocaleString()} MANA
```

Comment:
Same as above, should we be using `lib/utils#formatMana` for this line and the ones below?